Source/WebKit/UIProcess/Cocoa/UserMediaCaptureManagerProxy.cpp

```diff
@@ -410,6 +411,9 @@ class UserMediaCaptureManagerProxySourceProxy final
         return !m_isEnded;
     }
 
+    Ref<GenericPromise> protectedPendingAction() const { return m_pendingAction; }
+    Ref<RealtimeMediaSource> protectedSource() const { return m_source; }
```

Comment:
Might be worth checking the coding safe rules, but I wonder if using const Ref<> m_source (m_source is only set in the constructor) would allow us to directly use m_source instead of localling using protectedSource()?

Reply:
We need a smart pointer on the stack as per the rules so using a protected function is correct.